test/FunctionalTests/Microsoft.OData.Core.Tests/IntegrationTests/Reader/JsonLight/PropertyAndValueJsonLightReaderIntegrationTests.cs

```diff
@@ -773,6 +773,82 @@ public void ReadingNullValueForDeclaredCollectionPropertyInComplexTypeShouldFail
             read.ShouldThrow<ODataException>().WithMessage("A null value was found for the property named 'CountriesOrRegions', which has the expected type 'Collection(Edm.String)[Nullable=False]'. The expected type 'Collection(Edm.String)[Nullable=False]' does not allow null values.");
         }
 
+        [Fact]
+        public void ReadingNullValueForNonNullableCollectionofComplexShouldThrow()
+        {
+            EdmModel model = new EdmModel();
+            EdmComplexType complexType = new EdmComplexType("NS", "Car");
+            EdmComplexType door = new EdmComplexType("NS", "Door");
+            EdmComplexTypeReference frontDoor = new EdmComplexTypeReference(complexType, false);

```

Comment:
>frontDoor [](start = 36, length = 9)

The name of "frontDoor" sounds like a sub class name for "Door", but it looks not. Can you rename it?